sys/openbsd/buildlet.sh

```diff
@@ -0,0 +1,160 @@
+#!/bin/sh
+
+# Copyright 2018 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in
+# the LICENSE file.
+
+# Produces GCE image of syz-ci running on OpenBSD.
+
+# Mostly derived from Go buildlet generator with blessing from bradfitz@.
+
+set -eu -o pipefail
+
+readonly MIRROR="${MIRROR:-cloudflare.cdn.openbsd.org}"
+readonly VERSION="${VERSION:-6.4}"
+readonly DOWNLOAD_VERSION="${DOWNLOAD_VERSION:-snapshots}"
+readonly RELNO="${2:-${VERSION/./}}"
+
+# The only supported setting.
+readonly ARCH="amd64"
+
+readonly ISO="install${RELNO}-${ARCH}.iso"
+readonly ISO_PATCHED="install${RELNO}-${ARCH}-patched.iso"
+
+if [[ ! -f "${ISO}" ]]; then
+  curl -o "${ISO}" "https://${MIRROR}/pub/OpenBSD/${DOWNLOAD_VERSION}/${ARCH}/install${RELNO}.iso"
+fi
+
+# Create custom siteXX.tgz set.
+mkdir -p etc
+cat >install.site <<EOF
+#!/bin/sh
+set -x
+syspatch
+pkg_add -iv bash git go
+
+echo 'set tty com0' > boot.conf
+EOF
+
+cat >etc/installurl <<EOF
+https://${MIRROR}/pub/OpenBSD
+EOF
+cat >etc/rc.local <<EOF
+(
+  set -x
+
+  echo "starting syz-ci"
+)
+EOF
+chmod +x install.site
+tar -zcvf site${RELNO}.tgz install.site etc/{installurl,rc.local}
+
+# Autoinstall script.
+cat >auto_install.conf <<EOF
+System hostname = buildlet.syzkaller
+Which network interface = vio0
+IPv4 address for vio0 = dhcp
+IPv6 address for vio0 = none
+Password for root account = root
+Public ssh key for root account = ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBJeLpmSq+Dgrk01Ht+QtY0GDsY2gcwhT12SqizmacAF67TCa0n8OcpjVOpiiurb01Aa5lcl2WbakUFYmsp1U1l8=
```

Comment:
What are these keys? I think we need to generate a new pair for the image and use it, and produce private key as another artifact of the build. See tools/create-gce-image.sh for an example.

Reply:
In the current setup it does not even make sense to keep them secret, because anybody can ssh with syzkaller/syzkaller and then su into root with root password.

Reply:
I was planning to disallow all non-public key logins making passwords irrelevant. I think I did it for root, will do it universally. SGTY?

Reply:
Done.